packages/twenty-front/src/modules/workflow/workflow-steps/workflow-actions/ai-agent-action/hooks/__tests__/useAgentChat.test.ts

```diff
@@ -0,0 +1,651 @@
+import { AgentChatMessageRole } from '@/workflow/workflow-steps/workflow-actions/ai-agent-action/constants/agent-chat-message-role';
+import { act, renderHook } from '@testing-library/react';
+import { RecoilRoot } from 'recoil';
+import { isDefined } from 'twenty-shared/utils';
+import { AgentChatMessage } from '~/generated-metadata/graphql';
+import { useAgentChat } from '../useAgentChat';
+
+const mockSetAgentChatMessages = jest.fn();
+
+let mockUseRecoilState: jest.Mock;
+let mockSetAgentChatInput: jest.Mock;
+let mockUseScopedHotkeys: jest.Mock;
+let mockUseRecoilComponentStateV2: jest.Mock;
+let mockUseScrollWrapperElement: jest.Mock;
+let mockSendMessage: jest.Mock;
+let mockUseSendAgentChatMessageMutation: jest.Mock;
+let mockUseAgentChatThreadsQuery: jest.Mock;
+let mockUseAgentChatMessagesQuery: jest.Mock;
+
```

Comment:
Let's be careful with test that lock a certain way of doing things, they seem to test small things that don't necessarily have much value and will make the life of the next person refactoring this harder. Typically integration tests would bring more value